lib/resetState.ts

```diff
@@ -0,0 +1,61 @@
+type SerializableObjVal =
+  | SerializableObj
+  | string
+  | number
+  | boolean
+  | null
+  | SerializableObjVal[];
+
+type SerializableObj = {
+  [key in string]: SerializableObjVal;
+};
+
+let stateId = 0;
+function genStateId(): number {
+  stateId += 1;
+  return stateId;
+}
+
+const STATE_ID_PROPERTY = "___testStateStateId";
+
+let stateIdToState: Map<number, SerializableObj> = new Map();
```

Comment:
Why do you define this variable at the module level? Wouldn't it be better to define here the `type` only and the variable would be defined at the function level?

Reply:
Oh, ok I get it. WDYT about changing the name from singular to plural? Or like `stateIdToStateMap`

Reply:
Oh, ok I see. WDYT about renaming it from singular to plural? Or to something like `STATE_ID_TO_STATE_MAP` to indicate that it's global?

Reply:
renamed as suggested - there needs to be a global registry so we can track init states